app/components/MarketBar.tsx

```diff
@@ -1,53 +1,59 @@
 
-    if (isLoading) {
-     return <div>Loading...</div>;
-    }
-
   return (
-    <div className="flex items-center justify-center h-full gap-10 p-3 px-10 text-white bg-purple-text">
-      <div className="flex gap-2">
-        <span className="flex items-center h-full gap-1 font-thin">
-          <RiCoinsFill />
-          Coins
-        </span>{" "}
-        {data.data.active_cryptocurrencies}
-      </div>
-      <div className="flex items-center gap-2">
-        <GoTriangleUp className="text-green-300" />{" "}
-        {new Intl.NumberFormat("en-US", {
-          maximumSignificantDigits: 3,
-          style: "currency",
-          currency: `${currency}`,
-          notation: "compact",
-          compactDisplay: "short",
-        }).format(data.data.total_market_cap[currency])}
-      </div>
-      <div className="flex items-center gap-2">
-        {new Intl.NumberFormat("en-US", {
-          maximumSignificantDigits: 3,
-          style: "currency",
-          currency: `${currency}`,
-          notation: "compact",
-          compactDisplay: "short",
-        }).format(data.data.total_volume[currency])}
-      </div>
-      <div className="flex items-center gap-2">
-        {Math.floor(data.data.market_cap_percentage.btc)}%
-      </div>
-      <div className="flex items-center gap-2">
-        {Math.floor(data.data.market_cap_percentage.eth)}%
+    <div>
+      {isLoading && <div>Loading...</div>}
+      {data && (
+        <div className="flex items-center justify-center h-full gap-10 p-3 px-10 text-white bg-purple-text text-sm">
+        <div className="flex gap-2">
+          <span className="flex items-center h-full gap-1 font-thin">
+            <RiCoinsFill />
+            Coins
+          </span>{" "}
+          {data.data.active_cryptocurrencies}
+        </div>
+        <div className="flex items-center gap-2">
+          <GoTriangleUp className="text-green-300" />{" "}
+          {new Intl.NumberFormat("en-US", {
+            maximumSignificantDigits: 3,
+            style: "currency",
+            currency: `${currency}`,
+            notation: "compact",
+            compactDisplay: "short",
+          }).format(data.data.total_market_cap[currency])}
+        </div>
+        <div className="flex items-center gap-2">
+          {new Intl.NumberFormat("en-US", {
+            maximumSignificantDigits: 3,
+            style: "currency",
+            currency: `${currency}`,
+            notation: "compact",
+            compactDisplay: "short",
+          }).format(data.data.total_volume[currency])}
+          <ProgressBar percent={Math.floor((data.data.total_volume[currency]/data.data.total_market_cap[currency])*100)} fillColor="#ffffff" barColor="#8686a9" />
+        </div>
+        <div className="flex items-center gap-2">
+          {Math.floor(data.data.market_cap_percentage.btc)}%
+          <ProgressBar percent={Math.floor(data.data.market_cap_percentage.btc)} fillColor="#ea973d" barColor="#8686a9" />
```

Comment:
you can extract this logic into a component

Reply:
I don't understand which part you want me to put into a component, all I can think of is to put that percentage logic into a separate function which I will definitely do but let me know if that's what you meant or if there was something else you wanted me to change